xio-core/src/main/java/com/xjeffrose/xio/http/Client.java

```diff
@@ -65,4 +69,38 @@ public ChannelFuture write(Request request) {
       return channel.writeAndFlush(request).addListener(writeListener);
     }
   }
+
+  public CompletableFuture<ClientChannelResponse> connect() {
+    val outerResult = new CompletableFuture<ClientChannelResponse>();
+    internalConnect()
+        .addListeners(
+            connectionListener,
+            (resultFuture) -> {
+              val response =
+                  new ClientChannelResponse(
+                      resultFuture.isDone(), resultFuture.isSuccess(), resultFuture.cause());
+              outerResult.complete(response);
+            });
+    return outerResult;
```

Comment:
Why can't we return the ChannelFuture from `internalConnect`?

Reply:
actually in this case we can definitly return the channel future, the issue i think was more on the write(Request request) because if the channel is null you can't new ChannelPromise() and resolve it with a throwable cause

Reply:
I can change the connect() method to return the ChannelFuture